AbletonMCP_Remote_Script/handlers/automation.py

```
@@ -0,0 +1,345 @@
+            ctrl.log_message("Error listing automated params: " + str(e))
+        raise
+
+
+# --- New: Track-level automation and arrangement time editing (from MacWhite) ---
+
+
+def create_track_automation(song, track_index, parameter_name, automation_points, ctrl=None):
+    """Create automation for a track parameter (arrangement-level)."""
+    try:
+        if track_index < 0 or track_index >= len(song.tracks):
+            raise IndexError("Track index out of range")
```

Ruff (0.14.14)

[warning] Abstract `raise` to an inner function

(TRY301)